packages/edit-site/src/components/add-new-template/utils.js

```diff
@@ -98,33 +98,31 @@ const usePublicTaxonomies = () => {
 	}, [ taxonomies ] );
 };
 
-function usePostTypeNeedsUniqueIdentifier( publicPostTypes ) {
```

Comment:
Why not adjust the logic here?

Reply:
Until now, the same could be reused in `usePostTypeArchiveMenuItems()` and `usePostTypeMenuItems()`, I guess that's why this function existed, instead of having it inline like in [`useTaxonomiesMenuItems()`](https://github.com/WordPress/gutenberg/blob/1f6fec6575ca4e9bcd64822b533df9f3057b9ba1/packages/edit-site/src/components/add-new-template/utils.js#L347-L353).

With the changes in this PR, though, the same code can no longer be shared between `usePostTypeArchiveMenuItems()` and `usePostTypeMenuItems()` and because of that I removed the function and inlined the code in each respective place.

I don't have a strong opinion, so I'm happy to create functions for each one of them if you think that will keep the code cleaner.